Merge Docs and Release Notes subteams (#2665)
kubernetes/sig-release: release-team/role-handbooks/docs/README.md

    - [Prerequisites for Shadows](#prerequisites-for-shadows)
    - [Machine and GitHub Setup](#machine-and-github-setup)
      - [Set up krel](#set-up-krel)
      - [Fork the kubernetes repositories](#fork-the-kubernetes-repositories)
- [Release Timeline](#release-timeline)
- [Tools](#tools)
- [Debugging Tips](#debugging-tips)
- [Release Notes File Structure](#release-notes-file-structure)
- [TODOs](#todos)

## Overview

This document covers the responsibilities, time commitments, and timeline for Docs Leads shepherding docs releases for Kubernetes, including the generation and fine-tuning of Release Notes. Docs Lead Shadows should also read through this document and understand the launch processes so they can do it in the future.

The Docs Lead will be responsible for introducing shadows to the team and the release notes subcommand in [krel](https://github.com/kubernetes/release/tree/master/docs/krel). Shadows should expect to perform this task at least once themselves. The Docs Lead should indicate pain points and known issues to the shadows (if there are any) and work on strategies for overcoming them to avoid their coalescence during the later weeks.

> If there are potential fixes to the issues indicated and team members are keen, fixes and automation of the process is very welcome but not expected.

Reviewer: ```suggestion
If there are potential fixes to the issues indicated and team members are keen, fixes and automation of the process are encouraged but not required.
```

- Between 1 and 2 hours a week to attend the majority of Release Team (weekly) and Burndown meetings (daily during Code Freeze), subject to time zone appropriateness

> - Up to 1 hour weekly to attend [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs#meetings) for status reports

Reviewer 2: Can we add a bullet point to highlight the final release notes PR creation and reaching out to the sigs (via slack and email) to ask them to review the final draft?
 ```suggestion
- Up to 1 hour weekly to attend [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs#meetings) for status reports
- Create [known issues issue](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/release-notes/known-issues-bucket.md) in kubernetes/kubernetes to capture known issues for the release
- Send [an email to SIG-leads](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/release-notes/sig-leads-email.md) to ensure major changes for their SIGs are accurately reflected in the release notes
- Send a slack message to the sig channels to ensure major changes for the SIGs are accurately reflected in the release notes 
```

#### Onboarding Session (week 1) ~1 hour

> In the first week of the release cycle, the Docs Lead will organize an onboarding session with the shadows to go over general responsibilities and expectations.

Reviewer 2: Should we include going over krel as part of the onboarding session? 

Also @satyampsoni was creating a assignment sheet useful or is this something we should drop? https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/release-notes#week-0

#### Early and mid release cycle (weeks 1-8) ~1-5 hours/week

In the first 8 weeks of the cycle, the Docs team must attend weekly release meetings and run the [release-notes subcommand of krel](https://github.com/kubernetes/release/blob/master/docs/krel/release-notes.md) for every `alpha`, `beta` and `rc` to create an early draft of the release notes. This ensures that the overall quality of the release notes can be verified from the beginning of the release cycle.

Weekly branch syncs must be run to ensure consistency from `main` to `dev-[future-release]`. The Docs team is responsible for the communication of documentation-related deadlines to KEP owners and SIG leads, as well as the tracking of documentation and enforcement of deadlines.

All members of the Docs subteam should participate in PR reviews as time allows.

#### Late release cycle (weeks 9-12+) ~4-10 hours/week

This period has an increase in release team meetings each week and there is also significantly more work to do to ensure the release notes and documentation are in good working order for the release.

> The `release-notes` subcommand of `krel` must continue to be run on the release branch (for `beta` and `rc` releases) in order to pull in any outstanding PRs that are merged between the beginning of code freeze and the release.

Reviewer 2: Is this missing `alpha` intentionally? 
```suggestion
The `release-notes` subcommand of `krel` must continue to be run on the release branch (for `alpha`, `beta` and `rc` releases) in order to pull in any outstanding PRs that are merged between the beginning of code freeze and the release.
```

Docs Lead Shadows are people who are preparing to be a Docs Lead in the future. In addition to the time requirements above, shadows must:

- Strong written and verbal communications skills
- A working knowledge of Kubernetes concepts

> - Project management experience is helpful but not required

Reviewer: ```suggestion
- Familiarity with Git and command line tools.
- Project management experience is helpful but not required.
```

- Have signed the [contributor CLA](https://github.com/kubernetes/community/blob/master/CLA.md) for Kubernetes.
- Be invested in becoming an org member within the release cycle. This can often be achieved during the release cycle with sponsorship from a role lead. See the [Release Team onboarding guide](/release-team/release-team-onboarding.md) for more details.
- General knowledge of our SIG Docs [areas of responsibility](https://github.com/kubernetes/community/tree/master/sig-docs#subprojects).
- Experience with the general process involved with [contributing](https://kubernetes.io/docs/contribute/start/) to Kubernetes website.
- Be a [milestone maintainer](https://github.com/orgs/kubernetes/teams/website-milestone-maintainers/) in order to have the ability to add a milestone to an issue. Access can be requested by creating a [PR](https://github.com/kubernetes/org/pull/2235) against `kubernetes/org` repo.

### Machine and GitHub Setup

#### Setup krel

> [Install Go](https://golang.org/doc/install) in your machine and follow the [instructions to build the release tools](https://github.com/kubernetes/release/tree/master/docs/krel#installation) in your machine. Check the system requirements in the krel documentation.

Reviewer 3: We should also mention that`jq` is a hard requirement for running the `krel` command. This is not documented anywhere, and my fellow shadows encountered issues simply because they didn’t have `jq` installed.
@satyampsoni @npolshakova

Reviewer 2: @rayandas did you ever track down what the error was?

Reviewer 3: @npolshakova I didn’t. But I will try to regenerate it today. And let you know.

Reviewer 3: I couldn’t regenerate it on my system. I will keep an eye on errors faced by new contributors around this and then report it with details.

 - [`kubernetes/sig-release`](https://github.com/kubernetes/sig-release): This is where you will push regular PRs to keep the Release Notes draft up to date.
 - [`kubernetes-sigs/release-notes`](https://github.com/kubernetes-sigs/release-notes): This repo has the [release notes website](https://relnotes.k8s.io) sources

## Release Timeline

Reference the Docs [Release Timeline](Release-Timeline.md) for key dates and responsibilities during the release cycle and the Kubernetes Release Information page for the specific release (e.g. [Kubernetes 1.28 Release Information page](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.28/README.md) for information regarding the current release cycle including important dates, Release Team contact information, tracking spreadsheets, and more.

## Tools
- [krel](https://github.com/kubernetes/release/tree/master/docs/krel) The Kubernetes Release Toolbox *(note: always use the latest version of krel to ensure you have the latest fixes/patches)*
- [The krel `release-notes` subcommand](https://github.com/kubernetes/release/blob/master/docs/krel/release-notes.md)

> - [The old release notes tool](https://github.com/kubernetes/release/tree/master/cmd/release-notes)

Reviewer 2: Do we want to include a reference to the old release notes tool here? I don't think it's used by the release notes team anymore. If we want to include historical context, maybe we should link to Marko's talk on the release tooling: https://www.youtube.com/watch?v=ILpvdVDcSTU

Author: If it's no longer used at all, even in edge cases, it doesn't need to be included and I'll remove it. Thanks!

Reviewer 2: Yep, it's not used by shadows creating the release notes so I think we can remove it.

> - [Release notes website](https://relnotes.k8s.io) *(note: release notes website is only required for the final release)*

Reviewer 2: Everything since https://github.com/kubernetes/release/issues/1087 went in is handled automatically for the release notes website update. Once the json is available in the bucket, the website should just update. There aren't any manual steps the release team needs to do to update the site. 

Maybe we can change the note to reflect there's no manual steps required?

```suggestion
- [Release notes website](https://relnotes.k8s.io) *(note: release notes website is automatically updated)*
```

- [go-modiff](https://github.com/saschagrunert/go-modiff)
- [Hackmd](https://hackmd.io/)

> - [LWKD](http://lwkd.info) *(note: consider contributing to LWKD as part of your role)*

Reviewer: I wasn't aware that contribution for LWKD is part of this role. Is this still the case?

contains all the correctly formatted release notes text for the 1.30 release. The [JSON file](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.30/release-notes/release-notes-draft.json)
contains the release notes metadata that is used to generate the markdown file. 

When a Docs team member runs the `krel release-notes` command, a new session is created so that you can pause and resume
the editing process. For example the 1.30 release notes sessions are stored in the [sessions](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.30/release-notes/sessions)
directory in the sig-release repo under `release-1.30`.

If a Docs team member finds a mistake in the release notes, the edit will be saved as a map yaml file in the [maps](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.30/release-notes/maps)
directory. These maps are used to generate the markdown file and JSON file with the correctly edited release note. 

## TODOs

As a Docs shadow, if you are interested in contributing to the improvement of the release notes process, consider the following
areas of improvement:

> #### Github Workflow to Detect Common Release Note Issues

Reviewer 2: @Vyom-Yadav is actively working on adding this! 😄 https://github.com/kubernetes-sigs/release-actions/pull/101 Once it goes in, we should remove this section and add a section on reading the errors reported from the GitHub workflow.

Author: @Vyom-Yadav Should this be left in for this release or can it be removed?

Reviewer 2: Let's leave it for now. We can remove it once the workflow merges in sig-release
